histomicstk/segmentation/label/delete_overlap.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from .condense import condense
+from .delete import delete
+
+
+def delete_overlap(im_label, overlap_info):
+    """
+    Deletes overlapping regions from an image label based on overlap information and tile size.
+
+    Args:
+        im_label (ndarray): Image label represented as a NumPy array.
+        overlap_info (dict): Dictionary containing overlap information.
+                            It should have the following keys:
+                            'left', 'right', 'top', and 'bottom',
+                            each specifying the overlap amount in pixels.
+
+    Returns:
+        ndarray: Image label with overlapping regions deleted.
+    Note:
+        This function assumes the necessary imports, such as `np`, are already present.
+
+    """
+
+    # Compute the half of the overlap values
+    left_overlap = max(overlap_info['left'] // 2, 0)
+    right_overlap = max(overlap_info['right'] // 2, 0)
+    top_overlap = max(overlap_info['top'] // 2, 0)
+    bottom_overlap = max(overlap_info['bottom'] // 2, 0)
+
+    im_label_del = np.zeros_like(im_label)
```

Comment:
We should move this to just before it is used.  No need to allocate memory until it is useful.

Reply:
Great observation, I've made the changes according to your comments. The same allocation also happened in the `delete_border.py` code. So, I've also changed that too.